Source/Settings.cs

```diff
@@ -182,5 +182,13 @@ internal static void Initialize()
                 }
             }
         }
+
+        public static bool IsHeadwear(ApparelProperties apparelProperties)
+        {
+            return apparelProperties.LastLayer == ApparelLayerDefOf.Overhead
+                || apparelProperties.bodyPartGroups.Contains(BodyPartGroupDefOf.FullHead)
+                || apparelProperties.bodyPartGroups.Contains(BodyPartGroupDefOf.UpperHead)
+                || apparelProperties.bodyPartGroups.Contains(BodyPartGroupDefOf.Eyes);
+        }

```

Comment:
I'm concerned here as this is doing 3 separate queries against the same List while rendering. This will be considerably slower than iterating using a for loop (not a foreach loop as that'll be half as slow as a for loop)

I'd recommend writing this as
```
public static bool IsHeadwear(ApparelProperties apparelProperties)
{
    if (apparelProperties.LastLayer == ApparelLayerDefOf.Overhead)
        return true;
    for (int i = 0; i < apparelProperties.bodyPartGroups.Count; ++i)
    {
        var g = apparelProperties.bodyPartGroups[i];
        if (g == BodyPartGroupDefOf.FullHead || 
           g == BodyPartGroupDefOf.UpperHead ||
           g == BodyPartGroupDefOf.Eyes)
          {
              return true;
          }
      }
      return false;
}
```